creator-node/src/middlewares/contentAccess/contentAccessMiddleware.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  sendResponse,
+  errorResponseServerError,
+  errorResponseForbidden,
+  errorResponseUnauthorized,
+  errorResponseBadRequest,
+  CustomRequest
+} from '../../apiHelpers'
+import { NextFunction, Request, Response } from 'express'
+import type Logger from 'bunyan'
+import { checkContentAccess } from '../../contentAccess/contentAccessChecker'
+import config from '../../config'
+import { tracing } from '../../tracer'
+
+/**
+ * Middleware to validate requests to get content.
+ * For streaming, CN gets a DN signature:
+ * - generated by a DN with the cid along with a timestamp,
+ * This middleware will recover the DN signature and verify that it is from a registered DN.
+ * It also verifies that cid is same as those in the DN-signed data, and also verify that DN-signed data
+ * timestamp is relatively recent.
+ * If all these verifications are successful, then this middleware will proceed with the request as normal.
+ */
+export const contentAccessMiddleware = async (
+  request: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  const req = request as CustomRequest
+  const cid = req.params?.CID
+  if (!cid) {
+    return sendResponse(
+      req,
+      res,
+      errorResponseBadRequest(`Invalid request, no CID provided.`)
+    )
+  }
+
+  if (!config.get('contentAccessMiddleware')) {
```

Comment:
this config is not defined in config.js and also, why have it?

Reply:
This was updated and a typo

Reply:
yep i figured some comments would be outdated per my review comment :)

but also to my second question above, why have this config at all?

Reply:
I removed it